wordpress-mobile/WordPress-iOS
# Notification Likes: purge likes older than 7 days

Fixes #n/a

This adds a new `dateFetched` property to `LikeUser`. After new post or comment likes are fetched, any like that was last fetched at least 7 days ago is deleted.

To test:

To see that "old" likes are deleted, you can set the `dateFetched` to some time ago when `LikeUser` is created [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/c5a4a79b9012984084c3ca8fce2cedbde564f266/WordPress/Classes/Services/LikeUserHelpers.swift#L26). Example: 
`likeUser.dateFetched = Calendar.current.date(byAdding: .day, value: -14, to: Date()) ?? Date()`

In `purgeStaleLikes`, you can set a breakpoint to inspect the contents of `users` [fetched](https://github.com/wordpress-mobile/WordPress-iOS/blob/c5a4a79b9012984084c3ca8fce2cedbde564f266/WordPress/Classes/Services/LikeUserHelpers.swift#L68) (or log out `users`) .  

- Go to Notifications > Likes.
- Select both Comment and Post notifications.
- Check the contents of `users` (as noted above) to ensure old likes are deleted.

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature incomplete.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature incomplete.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature incomplete.

PR submission checklist:
